tools/provision-vms.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+CLUSTER_NAME=${1:-testing}
+
+NUM_MASTERS=${2:-1}
+NUM_WORKERS=${3:-1}
+
+MASTER_PROV_MAC_PREFIX="52:54:00:82:68:4"
+MASTER_BM_MAC_PREFIX="52:54:00:82:69:4"
+
+WORKER_PROV_MAC_PREFIX="52:54:00:82:68:5"
+WORKER_BM_MAC_PREFIX="52:54:00:82:69:5"
+
+LIBVIRT_STORAGE_POOL="default"
+
+MASTER_VBMC_PORT_START=623
+WORKER_VBMC_PORT_START=624
+
+delete_vbmc() {
+    local name="$1"
+
+    if vbmc show "$name" > /dev/null 2>&1; then
+        vbmc stop "$name" > /dev/null 2>&1
+        vbmc delete "$name" > /dev/null 2>&1
+    fi
+}
+
+create_vbmc() {
+    local name="$1"
+    local port="$2"
+
+    vbmc add "$name" --port "$port" --username admin --password admin
+    #for k in {1..10}; do
+        # HACK: Do this over and over because if the vBMC was deleted, the
+        # delete command may have been stuck, leaving the vBMC in an alive/running
+        # state, which otherwise fools our script
+        vbmc start "$name" > /dev/null 2>&1
+        #sleep 1
+    #done 
+}
+
+for i in $(sudo virsh list --all | grep $CLUSTER_NAME | awk '{print $2}'); do
+    delete_vbmc "$i"
+    sudo virsh destroy $i
+    sudo virsh vol-delete $i.qcow2 --pool=$LIBVIRT_STORAGE_POOL
+    sudo virsh undefine $i
+done
+
+for i in $(seq 1 "$NUM_MASTERS"); do
+    name="$CLUSTER_NAME-master-$i"
+
+    sudo virt-install --ram 16384 --vcpus 4 --os-variant rhel7 --cpu host-passthrough --disk size=40,pool=$LIBVIRT_STORAGE_POOL,device=disk,bus=virtio,format=qcow2 --import --noautoconsole --vnc --network=bridge:provisioning,mac="$MASTER_PROV_MAC_PREFIX$i" --network=bridge:baremetal,mac="$MASTER_BM_MAC_PREFIX$i" --name "$name" --os-type=linux --events on_reboot=restart --boot hd,network
```

Comment:
This is fine for now, but as an improvement, we could parametrize the system requirements (RAM, vCPUs, etc) and add those vars to common.sh, so the user with less powerful systems can change the flavor of the VMs.